### ISSUE TITLE
Failed to perform query to database after 10 attempts. Stop trying.

### ISSUE THREAD

**Issue by yerycs** [near/near-indexer-for-explorer#368]
I am getting following error when start near-indexer-for-explorer.
![image](https://user-images.githubusercontent.com/53403186/229676593-9bcf44c4-6477-4844-b7a1-a6a21ba063f9.png)
I am using local postgresql. 
When I check postgresql db, I can see first block height is indexed. So, there isn't issue in db credentials env.
What's the reason?

**Comment by yerycs** [comment]
When I remove `--non-strict-mode` argument, I don't see above error.
But It seems it stuck like following.
![image](https://user-images.githubusercontent.com/53403186/229683068-9e81fd03-502c-4ca6-a164-ffaac013f01a.png)
I think it should synced and It should show each block logs?

**Comment by khorolets** [comment]
The reasoning behind this is fairly simple.

When you use the --non-strict-mode option, some data is skipped entirely, and there are additional limitations on retries. Specifically, when using the non-strict mode, the indexer will give up after 10 retries, as you can see in your first screenshot, whereas in strict mode, retries are infinite.

If you intentionally chose to use --non-strict-mode, you can safely disregard those messages. They should eventually stop appearing or at least occur less frequently.

However, it's important to note that the Indexer for Explorer is designed to provide all data starting from the genesis block. While we added the non-strict mode as an option, the indexer was not intended to function that way, and it may behave unpredictably in non-strict mode.

Let me know if you have any further questions!

**Comment by yerycs** [comment]
This means, I need to run indexer explorer from the genesis block, it will run without issue?

**Comment by khorolets** [comment]
@yerycs Ideally, yes. If your intention is to replicate the same functionality that our instance was designed to have, then it would be best to follow the recommended usage.

I understand that it can be frustrating to use the tool as intended, and I would be happy to hear more about your use case. Based on that information, I can offer suggestions on how to achieve your goals while avoiding the need to run the Indexer for Explorer.

**Comment by yerycs** [comment]
Thank you. I am going to get transactions in Near. I can see public postgresql database for query. 
But I noticed `Danger` caution about usage of production. 
So, I think it is better to run this database on my side.
That's why I am going to run indexer-explorer.

**Comment by khorolets** [comment]
@yerycs I am sorry, the use case you've described is a bit ambiguous. 
1. Do you need Transactions only? What about the Receipts, Execution Outcomes?
2. Do you need all the Transactions or only the ones related to specific accounts/contracts?
3. Do you need something else?

**Comment by yerycs** [comment]
I need only transactions for specific accounts only.

**Comment by yerycs** [comment]
I don't like using other third party apis. I think Option 2 is good. One question, this indexer-explorer is also based in Near Lake Framework?

**Comment by khorolets** [comment]
Yes, it is. You can fork it and cut off everything you don't need. If you prefer this approach :)

**Comment by yerycs** [comment]
Thank you for your suggestions. :)
